src/app/debug/table/table.component.ts

```diff
@@ -181,42 +179,56 @@ export class TableComponent implements OnInit, OnDestroy {
   }
 
   retrieveRecords(): void {
-    this.doneRetrieving = false;
-    this.tableSettings.reportMetadata = [];
-    const httpServiceSubscription = this.httpService
-      .getMetadataReports(
-        this.tableSettings.displayAmount,
-        this.tableSettings.filterValues,
-        this.tableSettings.filterHeaders,
-        this.viewSettings.currentView.metadataNames,
-        this.viewSettings.currentView.storageName,
-      )
-      .subscribe({
-        next: (value) => {
-          this.setUniqueOptions(value);
-          this.tableSettings.reportMetadata = value;
-          this.tableSettings.tableLoaded = true;
-          this.toastService.showSuccess('Data loaded!');
-          this.doneRetrieving = true;
-          httpServiceSubscription.unsubscribe();
-        },
-        error: () => {
-          catchError(this.httpService.handleError());
-        },
-      });
+    if (this.viewSettings.currentView?.metadataNames && this.viewSettings.currentView?.storageName) {
+      this.doneRetrieving = false;
+      this.tableSettings.reportMetadata = [];
+      const httpServiceSubscription = this.httpService
+        .getMetadataReports(
+          this.tableSettings.displayAmount,
+          this.tableSettings.filterValues,
+          this.tableSettings.filterHeaders,
+          this.viewSettings.currentView.metadataNames,
+          this.viewSettings.currentView.storageName,
```

Comment:
We might add a convention for a Method with more than 2 parameters, but for now it's fine.